dependency-graph/dependency-review.yml

```diff
@@ -0,0 +1,11 @@
+name: 'Dependency Review'
+on: [pull_request]
```

Comment:
Our Action is not quite the same as those described in the PR template under `code-scanning` (the closest match IMO) our Action needs to run against PRs on push to topic branches, not `$default_branch`.

See the staff-ship repo linked in the PR description for details, and here's an example of a staff-ship installed repo where you can see the Action at work: [link](https://github.com/github/dependency-graph-api/actions/workflows/dependency-review.yml)

Reply:
Do you want this workflow for PR against any branch? That would be lots of run without much value. That's why we usually use `$default_branch` as that's the important branch.

Reply:
Yes we want this to run against any PR that's correct. The goal is for the Action to fail when a PR is introducing new known-vulnerable package versions in manifest file changes on the repo 👍

Reply:
We only want to run against `pull_request` so never default branch, only topic branch of PRs on the repo where the Action is installed 👍